src/devices/Ahtxx/AhtBase.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+using System.Threading;
+using UnitsNet;
+
+namespace Iot.Device.Ahtxx
+{
+    /// <summary>
+    /// AHT temperature and humidity sensor family.
+    /// Note: has been tested with AHT20 only, but should work with AHT10 and AHT15 as well.
+    /// Up to now all functions are contained in the base class, though there might be differences between the sensors types.
+    /// </summary>
+    public class AhtBase : IDisposable
+    {
+        /// <summary>
+        /// Address of AHTxx device (0x38). This address is fix and cannot be changed.
+        /// This implies that only one device can be attached to a single I2C bus at a time.
+        /// </summary>
+        public const int DeviceAddress = 0x38;
+
+        // datasheet version 1.1, table 10
+        private enum StatusBit : byte
+        {
+            Calibrated = 0x08,
+            Busy = 0x80
+        }
+
+        private enum Command : byte
+        {
+            Calibrate = 0xbe,
+            SoftRest = 0xba,
+            Measure = 0xac
+        }
+
+        private I2cDevice _i2cDevice = null;
+        private double _temperature;
+        private double _humidity;
+
+        /// <summary>
+        /// Initializes a new instance of the device connected through I2C interface.
+        /// </summary>
+        /// <paramref name="i2cDevice">Reference to the initialized I2C interface device</paramref>
+        public AhtBase(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+
+            // even if not clearly stated in datasheet, start with a software reset to assure clear start conditions
+            SoftReset();
+
+            // check whether the device indicates the need for a calibration cycle
+            // and perform calibration if indicated ==> c.f. datasheet, version 1.1, ch. 5.4
+            if (!IsCalibrated())
+            {
+                Calibrate();
+            }
+        }
+
+        /// <summary>
+        /// Gets the current temperature reading from the sensor.
+        /// Reading the temperature takes between 10 ms and 80 ms.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Temperature GetTemperature()
+        {
+            Measure();
+            return Temperature.FromDegreesCelsius(_temperature);
+        }
+
+        /// <summary>
+        /// Gets the current humidity reading from the sensor.
+        /// Reading the humidity takes between 10 ms and 80 ms.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Ratio GetHumidity()
+        {
+            Measure();
+            return new Ratio(_humidity, UnitsNet.Units.RatioUnit.Percent);
```

Comment:
maybe this should be better: `return Ratio.FromPercent(_humidity);`
Doesn't really change much but for the consistency with Temperature and how it's done on other bindings.

Reply:
Done.